prisma/schema.prisma

```diff
@@ -104,4 +106,29 @@ model Exercise{
   updated_at DateTime @updatedAt
   categories Category[]
   machine    Machine  @relation("ExerciseMachine", fields: [machine_id], references: [id])
+}
+model ContractCategory {
+  id         String   @id @default(uuid())
+  name       String   @unique
+  created_at DateTime @default(now()) 
+  updated_at DateTime @updatedAt
+  contracts  Contract[]
+}
+
+model Contract {
+  id              String          @id @default(uuid())
+  beneficiary_id  String
+  provider_id     String
+  start_date      DateTime
+  end_date        DateTime?
+  category_id     String
+  created_at      DateTime        @default(now())
+  updated_at      DateTime        @updatedAt
+  
+  beneficiary     User            @relation("BeneficiaryContracts", fields: [beneficiary_id], references: [id])
+  provider        User            @relation("ProviderContracts", fields: [provider_id], references: [id])
+  category        ContractCategory @relation(fields: [category_id], references: [id])
+
+  @@index([beneficiary_id], name: "idx_contract_beneficiary")
+  @@index([provider_id], name: "idx_contract_provider")
```

Comment:
no need for indexes right now. we will optimize the api later if needed

Reply:
Removed